src/nrnpython/nrnpython.cpp

```diff
@@ -175,20 +187,30 @@ extern "C" void nrnpython_start(int b) {
     PyOS_ReadlineFunctionPointer = nrnpython_getline;
 #endif
     // Is there a -c "command" or file.py arg.
+    bool python_error_encountered{false};
     for (i = 1; i < nrn_global_argc; ++i) {
       char* arg = nrn_global_argv[i];
       if (strcmp(arg, "-c") == 0 && i + 1 < nrn_global_argc) {
-        PyRun_SimpleString(nrn_global_argv[i + 1]);
+        if(PyRun_SimpleString(nrn_global_argv[i + 1])) {
+          python_error_encountered = true;
+        }
         break;
       } else if (strlen(arg) > 3 && strcmp(arg + strlen(arg) - 3, ".py") == 0) {
-        nrnpy_pyrun(arg);
+        if(!nrnpy_pyrun(arg)) {
+          python_error_encountered = true;
+        }
         break;
       }
     }
-    // In any case start interactive.
-    //		PyRun_InteractiveLoop(fopen("/dev/tty", "r"), "/dev/tty");
-    PyRun_InteractiveLoop(hoc_fin, "stdin");
+    // When running in non-interactive/batch mode then python_error_encountered
+    // imply NEURON exits with a nonzero code. In interactive mode, we can start
+    // a Python interpreter.
+    if(nrn_istty_) {
+      PyRun_InteractiveLoop(hoc_fin, "stdin");
+    }
+    return python_error_encountered;
   }
+  // olupton 2022-01-20: is this code ever reached?
```

Comment:
The first changeset of this repository ```1f6563798 [svn r1897] start development of 6.2``` in oc/hoc.c in ```hoc_quit() {``` just before ```exit(0)``` has the fragment
```
#if defined(USE_PYTHON)
        if (0 && (1 || use_python_interpreter)) {
                nrnpython_start(3);
                ret();
                pushx(0.);
                return;
        }
#endif
        exit(0); 
```
If sufficiently curious one could go back further into the history of that 'if' statement, but it seems the desire was to shut down graphics and then exit python. At any rate. let's just remove this entire fragment.

Reply:
Done!